app/views/tags/index.html.erb

```diff
@@ -21,8 +21,23 @@
 
 <main id="main-content" class="crayons-layout crayons-layout--header-inside">
   <header class="crayons-page-header">
-    <h1 class="crayons-title"><%= t("views.tags.heading") %></h1>
-    <%= link_to t("views.tags.following_tags"), dashboard_following_tags_path, class: "crayons-btn crayons-btn--ghost" %>
+    <h1 class="crayons-title">
+      <% if params[:q].present? %>
+        <%= t("views.tags.heading.search_results", q: params[:q]) %>
+      <% else %>
+        <%= t("views.tags.heading.default") %>
+      <% end %>
+    </h1>
+    <form accept-charset="UTF-8" method="get" action="<%= tags_path %>" role="search">
+      <div class="crayons-fields crayons-fields--horizontal">
+        <div class="crayons-field flex-1 relative">
+          <input class="crayons-header--search-input crayons-textfield js-search-input" type="text" id="nav-search" name="q" placeholder="<%= t("views.tags.search.placeholder") %>" autocomplete="off" />
```

Comment:
Just a small nitpick - since we don't have a visible label on the page, we should add an `aria-label` so screen reader users know what the input is for. I think it could just be the same as the placeholder, but maybe it's cleaner to create a separate i18n key for it

Reply:
added!